include/graaflib/graph.tpp

```diff
@@ -201,4 +215,29 @@ void graph<VERTEX_T, EDGE_T, GRAPH_TYPE_V>::remove_edge(
   std::abort();
 }
 
+template <typename VERTEX_T, typename EDGE_T>
+void get_transposed_graph(
+    const directed_graph<VERTEX_T, EDGE_T>& graph,
+    directed_graph<VERTEX_T, EDGE_T>& out_transposed_graph) {
+  for (const auto [edge_vertices, edge] : graph.get_edges()) {
+    const auto [vertex_id_lhs, vertex_id_rhs] = edge_vertices;
+
+    auto vertex_value_lhs = graph.get_vertex(vertex_id_lhs);
+    auto vertex_value_rhs = graph.get_vertex(vertex_id_rhs);
+    auto edge_type = edge;
+
+    if (!out_transposed_graph.has_vertex(vertex_id_lhs)) {
```

Comment:
What is the point of these checks? If the out graph already contains these vertices, shouldn't we throw an exception? Or better, could we construct a new empty graph in the function body and use that as the return value?

Then we can ensure ourselves that a vertex with that ID is not there yet

Reply:
> What is the point of these checks? If the out graph already contains these vertices, shouldn't we throw an exception? Or better, could we construct a new empty graph in the function body and use that as the return value?
> 
> Then we can ensure ourselves that a vertex with that ID is not there yet

I think what I was trying to get at here was a situation like this (in pseudocode):
```
NewGraph={}

Original Graph Edges = Edge(Vertex 1,  Vertex 2) and Edge(Vertex 2, Vertex3)

for Edge in Edges:
   NewGraph.Add(Edge.LeftVertex)
   NewGraph.Add(Edge.RightVertex)
```

In the above example, when we examine the second edge, we will encounter Vertex 2 again, because we have already added it when looping through the first edge. This is not an error, because we expect to find certain vertices again while looping if the edges are connected, so I'm not sure if using an exception makes sense here?  

But I definitely agree on constructing an empty graph as the return value!

Reply:
Ah yes, that makes sense then. I was thinking about how the code was also checking whether the passed in graph did not contain the IDs before we do any computation, but this will not be a problem once we construct the graph in the body of the function 👍🏻

Reply:
> Ah yes, that makes sense then. I was thinking about how the code was also checking whether the passed in graph did not contain the IDs before we do any computation, but this will not be a problem once we construct the graph in the body of the function 👍🏻

Great stuff! Those tweaks should now all be pushed. Would you like me to squash all the commits or are you happy for these to stay separate for the time being?